specifyweb/specify/tree_ranks.py

```diff
@@ -0,0 +1,345 @@
+from functools import wraps
+from hmac import new
+from operator import ge
+from enum import Enum
+
+from specifyweb.businessrules.exceptions import TreeBusinessRuleException
+from . import tree_extras
+from . import models as spmodels
+from sys import maxsize
+
+import logging
+logger = logging.getLogger(__name__)
+
+TAXON_RANKS = {
```

Comment:
"Shouting", LOL, THIS_IS_A_CONSTANT https://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations.  If it gets used in other files, I'll move it to it's own file, otherwise it's good here.  Specify6 is implemented similarly https://github.com/specify/specify6/blob/945d28f00288d2d64979848fb2fb63fda34bdd1b/src/edu/ku/brc/specify/datamodel/TaxonTreeDef.java#L48

Reply:
maybe be it was unclear from me. the variable is fine as all caps. don't see value in keeping rank names all caps. minor nitpick, so wouldn't block.